[Jaynon] iP

Jaynon opened the pull request:
# 🧽 Spongebob : A Chatbot for your daily needs. 

![alt text](https://i.imgur.com/IhjSup1.jpeg "Spongebot")
---

Spongebob fulfills **all** your needs for chatbot:

* 🅰️ Text-based
* 📖 Easy to learn
* ⌛ _Fast_ response time
* 🧳 Coded in Java

---

## ✨ Key Features

- [ ] **Manages** and **tracks** your tasks for you.
- [ ] Set different types of **tasks**, from deadlines to events.
- [ ] Set and manages **deadlines**.
- [ ] **Mark** tasks as done.
- [ ] **Filter** tasks based on description. 

--- 
## 🛠️ Setup

1. Download a version of Spongebob [here.](https://github.com/Jaynon/ip/releases)

2. Then, run the included jar file in any terminal with the line below.
```cmd
java -jar spongebob.jar
```

---
## 🧠 Contributions
Feel free to comment on this PR and suggest more features!

> **Requirements**
> <p> Have Java 17.0 and above</p>

Here is an snippet of the codebase:

```Java
public static void main(String[] args) {
        new Spongebob("data/spongebob.txt").run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Jaynon committed:
add greetings

Jaynon committed:
added echo

Jaynon committed:
added add, list

Jaynon committed:
added Mark as Done feature

Jaynon committed:
added ToDos, Events, Deadlines

Jaynon committed:
added automatic text ui testing

Jaynon committed:
added exception handling

Jaynon committed:
added delete

Jaynon committed:
added comments

Jaynon committed:
added enums

Jaynon committed:
Added Save

Jaynon committed:
Added comments

Jaynon committed:
added comments

Jaynon committed:
Merge branch 'branch-Level-7'

Jaynon committed:
added date

Jaynon committed:
Merge branch 'branch-Level-8'

Jaynon committed:
refactor into more OOP

Jaynon committed:
refactored into packages

Jaynon committed:
Merge branch 'add-gradle-support' into branch_gradle

Jaynon committed:
Merge branch 'branch_gradle'

Jaynon committed:
added JUnit tests

Jaynon committed:
Merge branch 'branch_JUnit'

Jaynon committed:
configured gradle to be able to build .jar files

Jaynon committed:
Merge branch 'branch_jar'

Jaynon committed:
Added JavaDoc Comments

Jaynon committed:
Update coding standard

Jaynon committed:
Add find keyword feature

Jaynon committed:
Merge branch 'branch-JavaDoc'

Jaynon committed:
Merge branch 'A-CodingStandard'

# Conflicts:
#	src/main/java/spongebob/command/AddCommand.java
#	src/main/java/spongebob/task/Event.java

Jaynon committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/spongebob/Parser.java
#	src/main/java/spongebob/command/FindCommand.java

Jaynon committed:
update README.md

Jaynon committed:
add stylecheck

Jaynon committed:
add GUI

Jaynon committed:
Merge branch 'branch-Level-10'

Jaynon committed:
Fix styling

Jaynon committed:
Update README

NgZiXin commented on src/main/java/spongebob/task/Event.java:
```diff
@@ -0,0 +1,80 @@
+    protected LocalDate to;
+
+    /**
+     * constructor for an event task, checks if description, start and end dates to be in correct format
```
In method header comments, the first sentence should start with a verb. Can be:  Construct an instance of Event after checking the format of description, start and end dates.

Jaynon replied:
Good point, i will update the description!

NgZiXin commented on src/main/java/spongebob/task/Event.java:
```diff
@@ -0,0 +1,80 @@
+
+    /**
+     * constructor for an event task, checks if description, start and end dates to be in correct format
+     * @param description description of task
```
Insert empty line between description and parameter section.

NgZiXin commented on src/main/java/spongebob/task/Event.java:
```diff
@@ -0,0 +1,80 @@
+    /**
+     * constructor for an event task, checks if description, start and end dates to be in correct format
+     * @param description description of task
+     * @param from  start date
```
Punctuation behind each parameter description.

NgZiXin commented on src/main/java/spongebob/command/FindCommand.java:
```diff
@@ -0,0 +1,50 @@
+    }
+
+    /**
+     * executes the command, prints out list of tasks that contains keyword
```
Remember that description should be a sentence. i.e. To start with a capital letter and end with a punctuation mark.

NgZiXin commented on src/main/java/spongebob/Parser.java:
```diff
@@ -0,0 +1,91 @@
+import spongebob.command.FindCommand;
+import spongebob.command.MarkCommand;
+import spongebob.command.UnknownCommand;
+/**
```
Line space above

NgZiXin commented on src/main/java/spongebob/command/AddCommand.java:
```diff
@@ -0,0 +1,73 @@
+            case "todo":
+                newTask = new Todo(arguments[1]);
+                break;
+
```
Can remove line space in btw.

NgZiXin commented on src/main/java/spongebob/task/Deadline.java:
```diff
@@ -0,0 +1,73 @@
+
+    }
+
+
```
Remove unnecessary line spaces.

NgZiXin reviewed (commented):
LGTM. Just code styling issues with Javadocs and random line spaces. For Javadocs, remember to write with full sentences (with capital letters and proper punctuation).

fymbc commented on src/main/java/spongebob/Parser.java:
```diff
@@ -0,0 +1,91 @@
+     */
+    public Command parse(String string) {
+
+        String[] command = string.split(" ", 2);
```
Since this variable will store multiple values as a string array, I suggest changing the name `command` to `commands`. What do you think?

Jaynon replied:
Yes, you are correct, that makes more sense!

fymbc commented on src/main/java/spongebob/task/Deadline.java:
```diff
@@ -0,0 +1,73 @@
+ */
+public class Deadline extends Task {
+
+    protected LocalDate by;
```
I think "by" may be slightly ambiguous though I'm aware this is the deadline date. Perhaps "deadline" might be a slightly more suitable variable name as it is a noun?

Jaynon replied:
Yep, I will change that too.

fymbc commented on src/main/java/spongebob/task/Event.java:
```diff
@@ -0,0 +1,80 @@
+ */
+public class Event extends Task {
+
+    protected LocalDate from;
+    protected LocalDate to;
```
I think "from"  and "to" may be slightly ambiguous though I'm aware these are the start and end dates. Perhaps "startDate" and "endDate" might be a slightly more suitable variable name as it is a noun?

fymbc reviewed (commented):
LGTM! Good naming! Just some slight suggestions :)

Jaynon committed:
Update styling and bug fixes

Styling has been updated based on review feedback,
Fixed a major bug where the previous command is not overwritten when calling the same command type.

Jaynon committed:
Added assertions

Storage must be loaded with a text file before executing commands.
Commands with different keywords such as "add" and "mark" should have the correct keywords when executing.

Jaynon committed:
Improve code quality

simplified parser and reduced nesting
parameter and variable names are more clear and understandable.

Jaynon committed:
Merge pull request #1 from Jaynon/branch-Assertions

Added assertions

Jaynon committed:
Merge branch 'branch-Assertions' of https://github.com/Jaynon/ip

* 'branch-Assertions' of https://github.com/Jaynon/ip:
  Added assertions

Jaynon committed:
Revert "Merge branch 'branch-Assertions' of https://github.com/Jaynon/ip"

This reverts commit 8482b019eb5429a928cf1cfd15073403d6bc8b87, reversing
changes made to 0d7cbf5976e3169c85e60d9c8865996e862f48b7.

Jaynon committed:
Reapply "Merge branch 'branch-Assertions' of https://github.com/Jaynon/ip"

This reverts commit 280bb934482e8f8649f12b444dbabbea12f384ee.

Jaynon committed:
Merge pull request #2 from Jaynon/branch-CodeQuality

Improve code quality

Jaynon committed:
Merge branch 'master' into branch-CodeQuality

# Conflicts:
#	data/spongebob.txt
#	src/main/java/spongebob/command/FindCommand.java
#	src/main/java/spongebob/command/MarkCommand.java

Jaynon committed:
Bug Fixes

Mark Command assertion is corrected

Jaynon committed:
Add Continuous Integration

Jaynon committed:
Create main.yml

Jaynon committed:
Merge branch 'master' of https://github.com/Jaynon/ip

Jaynon committed:
Delete extra yml file

Jaynon committed:
Add Tagging

use the 'tag {index} {name}' command to add a tag to tasks

Jaynon committed:
Merge branch 'branch-Tagging'

Jaynon committed:
Add JavaDoc Comments

TaskList and Command now have JavaDoc Comments

Jaynon committed:
Update README.md for User Guide

User Guide should now show how to use the bot.

Jaynon committed:
Add text file support

When packaged into a jar file, the bot can read from an external text
file and create one if the file does not exist.

Updated user guide to support installation instructions
